neural_structured_learning/examples/notebooks/adversarial_cnn_transfer_learning_fashionmnist.ipynb

```diff
@@ -0,0 +1,1263 @@
+        "\n",
+        "We will now leverage Neural Structured Learning (NSL) to train an adversarial-regularized VGG-19 model."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "SkAKUGxq9Wes"
+      },
+      "source": [
+        "# Install NSL Dependency"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "bBobTSU1J3-O"
+      },
+      "source": [
+        "!pip install neural-structured-learning"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "ZFSrRBf6Fdmr"
+      },
+      "source": [
+        "import neural_structured_learning as nsl"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "PTVVz5mH9aSH"
+      },
+      "source": [
+        "# Adversarial Learning Configs\n",
+        "\n",
+        "*   **`adv_multiplier`**: The weight of adversarial loss in the training\n",
+        "objective, relative to the labeled loss.\n",
+        "*   **`adv_step_size`**: The magnitude of adversarial perturbation.\n",
+        "*  **`adv_grad_norm`**: The norm to measure the magnitude of adversarial\n",
+        "perturbation.\n",
+        "\n",
+        "Adversarial Neighbors are created leveraging the above config settings.\n",
+        "\n",
+        "__`adv_neighbor = input_features + adv_step_size * gradient`__ where __`adv_step_size`__ is the step size (analogous to learning rate) for searching/calculating adversarial neighbors."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "tp1xpfqAJ268"
+      },
+      "source": [
+        "adv_multiplier = 0.45\n",
+        "adv_step_size = 0.95\n",
+        "adv_grad_norm = 'l2'\n",
+        "\n",
+        "adversarial_config = nsl.configs.make_adv_reg_config(\n",
+        "  multiplier=adv_multiplier,\n",
+        "  adv_step_size=adv_step_size,\n",
+        "  adv_grad_norm=adv_grad_norm\n",
+        ")"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "mdNwNBdxLfPR"
+      },
+      "source": [
+        "adversarial_config"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "RyprtLa9y1w5"
+      },
+      "source": [
+        "#### Feel free to play around with the hyperparameters and observe model performance"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "PBLHRAW497OF"
+      },
+      "source": [
+        "# Fine-tuning VGG-19 CNN Model with Adversarial Learning - Adversarial Model\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "nw01eSoU-BvZ"
+      },
+      "source": [
+        "## Create Base Model Architecture"
```

Comment:
This section seems to be repeated from the base model architecture definition earlier in the tutorial. Can you create a shared function that creates the base model, and just invoke it here again to instantiate a fresh copy of the base model?

Reply:
Actually since we are explaining how to build a model in parts in separate markdown sections at the beginning I didn't make this into a function as then would need to put all those sections under one heading, let me know if that makes sense then I can do it also

Reply:
Sure, let's leave this as is then.